source/blog/2016-03-04-atomic-site-on-atomic-app.md

```diff
@@ -0,0 +1,131 @@
+
+Because we're sharing some volumes and I have SELinux turned on (don't you?), I also need to tell SELinux that the container is allowed to share access to those directories. So I added a little [selinux_share.sh](https://github.com/projectatomic/atomic-site/blob/master/selinux_share.sh) script in the repo's home directory that sets those.
+
+The second part is to create a placeholder Dockerfile, which is used to build the atomic.app wrapper container.  These files pretty much all look the same:
+
+```
+FROM projectatomic/atomicapp:0.4.2
+
+MAINTAINER Red Hat, Inc. <container-tools@redhat.com>
+
+LABEL io.projectatomic.nulecule.providers="docker" \
+      io.projectatomic.nulecule.specversion="0.0.2"
+
+ADD /Nulecule /Dockerfile README.md /application-entity/
+ADD /artifacts /application-entity/artifacts
+```
+
+The only thing that really changes is which providers you support and the spec and app versions.
+
+Now we get to the good stuff: the Nulecule file, which is most of the definition. Let's do this one a little at a time.
+
+First, we define the spec version and the application ID in the header. Nothing exciting here:
+
+```
+---
+specversion: 0.0.2
+id: atomic-site
+```
+
+Next, we set some overall metadata, which is pretty much just for registry and future searching use:
+
+```
+metadata:
+  name: Atomic Site app
+  appversion: 0.1.0
+  description: Atomic app for testing the Atomic Website
+```
+
+The "graph" contains the definition of all of the app's objects. While ordering-independent, we're going to start it with the parameters which will need to be supplied by the answers.conf file:
+
+```
+graph:
+  - name: atomic-site
+    params:
+      - name: image
+        description: The webserver image
+        default: jberkus/atomic-site
+      - name: hostport
+        description: The host TCP port as the external endpoint
+        default: 4567
+      - name: sourcedir
+        description: location of the middleman source directory
+      - name: datadir
+        description: the middleman data directory
+```
+
+Here you'll notice that we supplied some defaults for the image name and the port to be used on the host, because we expect those to be the same for most users. We don't expect that most users will have the same path to their local copy of the atomic-site repo, though, so those parameters have no defaults, meaning that the user has to supply them.
+
+Finally, the "artifacts" section defines where the provider files live:
+
+```
+artifacts:
+  docker:
+    - file://artifacts/docker/atomic-site-run
+```
+
+Since we're supporting only Docker right now, one line is all we need. We also don't currently define any persistentVolumes, even though we'll be using some volumes; this is because of a limitation which will be resolved in Docker 1.10.
```

Comment:
there's an issue for this, but I can't find it right now.  Supposedly we'll be supporting the persistentVolume in Docker after 1.10 because there will be some kind of analgous support in Docker.